pilot/pkg/model/push_context.go

```diff
@@ -352,9 +363,12 @@ func (ps *PushContext) VirtualServices(proxy *Proxy, gateways map[string]bool) [
 
 // DestinationRule returns a destination rule for a service name in a given domain.
 func (ps *PushContext) DestinationRule(proxy *Proxy, hostname Hostname) *Config {
-	// TODO: use the proxy namespace to return only public destination rules if in different namespace
 	if c, ok := MostSpecificHostMatch(hostname, ps.destinationRuleHosts); ok {
```

Comment:
Is this correct ? MostSpecificHostMatch may return a private rule - but a less specific may be public.

Reply:
@costinm I think this name is confusing, actually it iterate `destinationRuleHosts` and return the matched host once it found.  Currently we do not support multi-host match.   

This is a good question, I will add a TODO